instance/models/instance.py

```diff
@@ -134,6 +152,68 @@ def status(self):
         else:
             return active_server_set[0].status
 
+    @property
+    def event_context(self):
+        """
+        Context dictionary to include in events
+        """
+        return {'instance_id': self.pk}
+
+    @staticmethod
+    def on_post_save(sender, instance, created, **kwargs):
+        """
+        Triggered by the post_save event, with `created` a boolean indicating if the object was just
+        created in the DB
+        """
+        self = instance
+        self.update_logger()
+
+    def update_logger(self):
+        """
+        Start referencing the instance in logs once the DB entry exist
+        """
+        if self.pk is not None:
+            self.logger = InstanceLoggerAdapter(logger, {'obj': self})
+
+    @property
+    def log_text(self):
+        """
+        Combines the instance and server log outputs in chronological order
+        Currently only supports one non-terminated server at a time
+        Returned as a text string
+        """
+        current_server = self.active_server_set.get()
+        # TODO: Filter out log entries for which the user doesn't have view rights
+        server_log_entry_set = current_server.log_entry_set.order_by('pk')\
+                                                           .iterator()
+        instance_log_entry_set = self.log_entry_set.order_by('pk')\
+                                                   .iterator()
+
+        next_server_log_entry = partial(next, server_log_entry_set, None)
+        next_instance_log_entry = partial(next, instance_log_entry_set, None)
+
+        log_text = ''
+        instance_log_entry = next_instance_log_entry()
+        server_log_entry = next_server_log_entry()
+
+        while instance_log_entry is not None and server_log_entry is not None:
+            if server_log_entry.created < instance_log_entry.created:
+                log_text += '{}\n'.format(server_log_entry)
+                server_log_entry = next_server_log_entry()
+            else:
+                log_text += '{}\n'.format(instance_log_entry)
+                instance_log_entry = next_instance_log_entry()
+
+        while instance_log_entry is not None:
+            log_text += '{}\n'.format(instance_log_entry)
+            instance_log_entry = next_instance_log_entry()
+
+        while server_log_entry is not None:
+            log_text += '{}\n'.format(server_log_entry)
+            server_log_entry = next_server_log_entry()
+
+        return log_text
```

Comment:
@antoviaque This method makes me wonder whether spreading the logs over multiple tables is the best approach. It might be worth considering having a single `log_entries(created, level, text, instance, server)` table where `server` and `instance` could be `NULL`.